iceoryx_hoofs/test/moduletests/test_buffer_stack.cpp

```diff
@@ -254,6 +258,34 @@ TEST_F(stack_test, CopyConstructorWorksAndCallsTestClassCopyConstructor)
     EXPECT_THAT(testStack.pop().value(), Eq(TestClass(ELEMENT, ELEMENT, ELEMENT)));
 }
 
+TEST_F(stack_test, CopyCtorWithOneElementLeadsToEqualCtorAndDtorCalls)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "dc44fcdc-ced6-4345-822b-58b9e58baf85");
+    constexpr uint32_t ELEMENT{37};
+    {
+        stack<TestClass, STACK_SIZE> other;
+        other.push(TestClass(ELEMENT, ELEMENT, ELEMENT));
+
+        stack<TestClass, STACK_SIZE> sut(other);
+
+        EXPECT_EQ(other.size(), 1);
+        EXPECT_EQ(sut.size(), 1);
+
+        EXPECT_THAT(TestClass::dTor, Eq(1));
+        EXPECT_THAT(TestClass::copyCTor, Eq(1));
+        EXPECT_THAT(TestClass::moveCTor, Eq(1));
+
+        auto element = sut.pop();
+        ASSERT_TRUE(element.has_value());
+        EXPECT_THAT(other.size(), Eq(1));
+        EXPECT_THAT(sut.size(), Eq(0));
+    }
+    EXPECT_THAT(TestClass::dTor, Eq(5));
+    EXPECT_THAT(TestClass::cTor, Eq(5));
```

Comment:
@elBoberido I have troubles to understand why the d'tor and c'tor call are one less each in case of the [ASAN job](https://github.com/eclipse-iceoryx/iceoryx/actions/runs/4175619432/jobs/7230810479). Wasn't able to reproduce this locally.

It also happens in `popCreatesSpaceForAnotherElement`.

Reply:
It's related to `clang-15` and [copy elision](https://en.cppreference.com/w/cpp/language/copy_elision).